CONTRIBUTING.md

```diff
@@ -1,26 +1,69 @@
+This project is written in [TypeScript](https://www.typescriptlang.org/), a typed variant of JavaScript, and we use [Prettier](https://prettier.io/) to get a consistent code style.
+
+Because of how GitHub Actions are run, the source code of this project is transpiled from TypeScript into JavaScript. The transpiled code (found in `lib/`) is subsequently compiled using [NCC](https://github.com/vercel/ncc/blob/master/readme.md) (found in `dist/`) to avoid having to include the `node_modules/` directory in the repository.
+
+## Submitting a pull request
+
+1. [Fork][fork] and clone the repository
 1. Configure and install the dependencies: `npm install`
 1. Create a new branch: `git checkout -b my-branch-name`
-1. Make your change and build the action using `npm run build`
-1. Push to your fork and [submit a pull request](https://github.com/actions/labeler/compare)
-1. Pat your self on the back and wait for your pull request to be reviewed and merged.
+1. Make your change, add tests, and make sure the tests still pass: `npm run test`
+1. Make sure your code is correctly formatted: `npm run format`
+1. Update `dist/index.js` using `npm run build`. This creates a single javascript file that is used as an entrypoint for the action
+1. Push to your fork and [submit a pull request][pr]
+1. Pat yourself on the back and wait for your pull request to be reviewed and merged.
 
 Here are a few things you can do that will increase the likelihood of your pull request being accepted:
 
-- Follow the coding style used in this project. You can do this by running `npm run format`.
+- Write tests.
 - Keep your change as focused as possible. If there are multiple changes you would like to make that are not dependent upon each other, consider submitting them as separate pull requests.
-- Write [good commit messages](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
+## Releasing a new version
+
+All the concepts from [the actions/toolkit release docs](https://github.com/actions/toolkit/blob/main/docs/action-versioning.md) apply. Please read that first!
+
+Once the changes are merged into main, a repo maintainer should:
+
+1. Bump the package version by running [`npm version [major|minor|patch]`](https://docs.npmjs.com/cli/v7/commands/npm-version). We adhere to [SemVer 2.0](https://semver.org/spec/v2.0.0.html) to the best of our ability. Commit the changes to `package.json` and `package-lock.json` and push them to main.
+1. [Draft a new release](https://github.com/actions/labeler/releases/new) pointing to the ref of the version bump you just made. Publish the release to the marketplace when complete.
+1. Finally: update the most recent "shorthand tag" (v1, v2, v3, etc) to point to the specific ref of the release you just made. For example, if we just released `v1.1.0`, we would rewrite the `v1` tag like this:
+
+```
+git tag -fa v1 v1.1.0 -m "Update v1 tag to point to v1.1.0"
+git push origin v1 --force
+```
+
```

Comment:
@TingluoHuang How does this sound to you?

Reply:
I would like to get @thboop to take a look since I haven't done lots of action releases.